src/main/java/uk/gov/hmcts/reform/cwrdapi/service/impl/CaseWorkerServiceImpl.java

```diff
@@ -0,0 +1,279 @@
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static java.util.Objects.requireNonNull;
+
+@Service
+@Slf4j
+@Setter
+public class CaseWorkerServiceImpl implements CaseWorkerService {
+
+
+    @Value("${loggingComponentName}")
+    private String loggingComponentName;
+
+    @Autowired
+    CaseWorkerProfileRepository caseWorkerProfileRepo;
+
+    @Autowired
+    RoleTypeRepository roleTypeRepository;
+
+    @Autowired
+    UserTypeRepository userTypeRepository;
+
+    @Autowired
+    CaseWorkerIdamRoleAssociationRepository cwIdamRoleAssocRepository;
+
+    @Autowired
+    private UserProfileFeignClient userProfileFeignClient;
+
+    List<RoleType> roleTypes = new ArrayList<>();
+
+    List<UserType> userTypes = new ArrayList<>();
+
+    CaseWorkerProfile caseWorkerProfile = null;
+
+
+    @Override
+    public ResponseEntity<Object> saveOrUpdateOrDeleteCaseWorkerUserProfiles(List<CaseWorkersProfileCreationRequest>
+                                                                           cwrsProfilesCreationRequest) {
+        List<CaseWorkerProfile> caseWorkerProfiles = new ArrayList<>();
+        try {
+            getRolesAndUserTypes();
+            cwrsProfilesCreationRequest.forEach(cwrProfileCreationRequest -> {
+                CaseWorkerProfile caseWorkerProfileRespose = null;
+
+                caseWorkerProfile = caseWorkerProfileRepo
+                        .findByEmailId(cwrProfileCreationRequest.getEmailId());
+                if (Objects.isNull(caseWorkerProfile)) {
+
+                    caseWorkerProfileRespose = createCaseWorkerProfile(cwrProfileCreationRequest);
+                    if (Objects.nonNull(caseWorkerProfileRespose)) {
+                        // collecting all the successfully case worker profiles to save in caseworker db.
+                        caseWorkerProfiles.add(caseWorkerProfileRespose);
+
+                    }
+                } else if (Objects.nonNull(caseWorkerProfile) && !caseWorkerProfile.getDeleteFlag()) {
+
+                    //update the existing case worker profile logic
+                } else if (caseWorkerProfile.getDeleteFlag()) {
+
+                    // updating the status in idam to suspend.
+                }
+            });
+
+            /**  caseworker profile bacth save
+             in caseWorkerProfile request contains all the associated entities information and while
+             saving it automatically saves data in the sub entities like cwlocation, cwWorkArea,cwRole and
+             caseworker profile and no need to explicitly invoke the save method for each entities.
+             */
+            caseWorkerProfileRepo.saveAll(caseWorkerProfiles);
+            log.info("{}::case worker profiles inserted::" + caseWorkerProfiles.size(), loggingComponentName);
+
+        } catch (Exception exp) {
+
+            log.error("{}:: createCaseWorkerUserProfiles failed" + exp, loggingComponentName);
```

Comment:
incorrect logging + not needed and please put {}, {}

Reply:
ok. i will change

Reply:
done